starlette_web/common/database/model_mixin.py

```diff
@@ -0,0 +1,241 @@
+    ) -> ScalarResult:
+        query = cls.prepare_query(limit=limit, offset=offset, **filter_kwargs)
+        result = await db_session.execute(query)
+        return result.scalars()
+
+    @classmethod
+    async def async_get(cls, db_session: AsyncSession, **filter_kwargs) -> "DBModel":
+        query = cls.prepare_query(**filter_kwargs)
+        result = await db_session.execute(query)
+        return result.scalars().first()
+
+    @classmethod
+    async def async_update(
+        cls, db_session: AsyncSession, filter_kwargs: dict, update_data: dict, db_commit=False,
+    ):
+        query = (
+            update(cls)
+            .where(cls._filter_criteria(filter_kwargs))
+            .values(**update_data)
+            .execution_options(synchronize_session="fetch")
+        )
+        await db_session.execute(query)
+        if db_commit:
+            await db_session.commit()
+
+    @classmethod
+    async def async_delete(cls, db_session: AsyncSession, filter_kwargs: dict, db_commit=False):
+        query = (
+            delete(cls)
+            .where(cls._filter_criteria(filter_kwargs))
+            .execution_options(synchronize_session="fetch")
+        )
+        await db_session.execute(query)
+        if db_commit:
+            await db_session.commit()
+
+    @classmethod
+    async def async_create(cls, db_session: AsyncSession, db_commit=False, **data) -> "DBModel":
+        instance = cls(**data)  # noqa
+        db_session.add_all([instance])
+
+        # Explicit flush() to populate newly created instance with primary key (if autogenerated).
+        await db_session.flush()
+
+        if db_commit:
+            await db_session.commit()
+        return instance
+
+    @classmethod
+    async def async_create_or_update(
+        cls,
+        db_session: AsyncSession,
+        filter_kwargs: dict,
+        update_data: dict,
+        db_commit=False,
+        force_update=False,
+        update_to_null=False,
+    ) -> "DBModel":
+        """
+        Allows to create entry, if not exists, or update an existing query.
+
+        :param db_session: active instance of AsyncSession
+        :param filter_kwargs: function will search entry by these conditions
+        :param update_data: function will update these fields.
+            "update_data" has priority over "filter_kwargs" in case of a new entry.
+        :param db_commit: bool, whether to execute db_session.commit()
+        :param force_update: bool, whether to do update is instance exists and has not been changed
+        :param update_to_null: bool, whether to update currently non-null values to new null values
+        :returns: instance of DBModel
+        """
+        # TODO: contrib.postgres.PostgresModelMixin with this method reloaded
+        #  (using on_conflict do update)
+        instance = await cls.async_get(db_session, **filter_kwargs)
+        if instance:
```

Comment:
у нас python 3.10 можно юзать так:
```suggestion
    if instance := await cls.async_get(db_session, **filter_kwargs):
```

Reply:
я ещё не привык к моржовому оператору...

Reply:
давай привыкать) удобная штука